woocommerce/woocommerce-ios
# Coupons: Duplicate product list selector view controller to prepare for the UI update

Part of #6489 


### Description

For the Coupons project, we need to modernize the UI for the product selector screen. Since we already have a screen for selecting products, it's probably wiser to update the old screen instead of creating a new one because the effort required for handling the product list is more significant than the UI updates.

Changes:
- Rename the old selector to `LegacyProductListSelectorViewController`
- Move the files relating to the product list selector to a separate group under the Products group in the project navigator to make it easier to find.
- Duplicate the selector view controller
- Switch to use the new selector view controller if the coupon edit feature flag is enabled in the `LinkedProductsListSelectorViewController`.
- Update the class of the `LinkedProductsListSelectorViewController.xib` file owner to `LinkedProductsListSelectorViewController`. I'm not sure why it still worked before with the `ProductListSelectorViewController` being the file owner 🤷‍♀️.

### Testing instructions

Please double-check that the product list selector in the linked product feature still works properly with and without the feature flag `couponEditing` enabled:
- Navigate to the Products tab and select a product.
- In the product form, select Add more details > Linked Products > Add Products > Add Products.
- Notice that the selector still works as before: selection, deselection, submission.

### Screenshots

The product selector screen should be as before:

<img src="https://user-images.githubusercontent.com/5533851/162164166-4b9796bc-b80d-4156-a1af-9f913335df16.png" width=320 />

---
